src/include/sof/audio/module_adapter/module/generic.h

```diff
@@ -42,7 +42,7 @@ static struct comp_dev *module_##adapter##_shim_new(const struct comp_driver *dr
 } \
 \
 static const struct comp_driver comp_##adapter##_module = { \
-	.type = SOF_COMP_MODULE_ADAPTER, \
+	.type = comp_type, \
```

Comment:
this is a rather significant change, and I think it will break at least this check: https://github.com/thesofproject/sof/blob/70272282dbfeba2e0df7653cb104647968d2f30b/src/audio/module_adapter/module_adapter.c#L135 and possibly more. Whereas my proposal to change `comp_specific_builder()` would only affect IPC3. @ranj063 do you think this is a good idea?

Reply:
@lyakh , but how would I know in comp_specific_builder() what kind of messages the module - found with uuid - actually accepts, unless its somehow built in to the module? I can of course add an extra member to struct comp_driver, but I do not think this issue can not be solved without adding the accepted componen message type to each module.

Reply:
No, don't think it's possible to check the type generically in common code for all components with the current module-adapter API. That's why I'm not proposing that. Instead I'm just proposing to (1) check that the type value in IPC is meaningful - one of the values, that the firmware recognises, and error out otherwise, instead of silently ignoring it, and (2) in each individual component driver check that the type from the IPC is correct

Reply:
Ok, simple enough.